[TM] Add spec for AndroidToast

wojteg1337 opened the pull request:
## Summary

part of #24875 

## Changelog

[General] [Added] - Add TM spec for AndroidToast

## Test Plan

Flow passes

lunaleaps commented on Libraries/Components/ToastAndroid/NativeToastAndroid.js:
```diff
@@ -0,0 +1,44 @@
+  +removeListeners: (count: number) => void;
+}
+
+export default TurboModuleRegistry.getEnforcing<Spec>('ToastAndroid');
```
I think we need to conditionally call `getEnforcing` here

See https://github.com/facebook/react-native/issues/24875
> In the case that a NativeModule is platform-specific, please use Platform.OS and conditionally call TurboModuleRegistry.getEnforcing.

ericlewis replied:
^ this is correct

lunaleaps commented on Libraries/Components/ToastAndroid/NativeToastAndroid.js:
```diff
@@ -0,0 +1,44 @@
+  ) => void;
+
+  // RCTEventEmitter
+  +addListener: (eventName: string) => void;
```
Is this an event emitter class?

ericlewis replied:
Please remove these methods, it is not an event emitter.

wojteg1337 commented:
@ericlewis i fixed comments you left, could you review it once again?

ericlewis commented:
@wojteg1337 there are tons of errors here still. Since the library has been changed to conditional you probably need to do invariant checks on wether or not the library exists at existing call sites. Once you get all the tests passing (except e2e, don’t worry about that) I’ll approve.

lunaleaps commented on Libraries/Components/ToastAndroid/NativeToastAndroid.js:
```diff
@@ -0,0 +1,43 @@
+
+export default (Platform.OS === 'android'
+  ? TurboModuleRegistry.getEnforcing<Spec>('ToastAndroid')
+  : null);
```
Is it odd to `export default null`? I wonder if we shoudn't export at all if iOS

ericlewis commented:
you have to export something, undefined or null. i dont think there is a specific on witch. 

Sent via Superhuman ( https://sprh.mn/?vip=ericlewis777@gmail.com )

On Fri, May 17, 2019 at 12:55 PM, Luna < notifications@github.com > wrote:

> 
> 
> 
> *@lunaleaps* commented on this pull request.
> 
> 
> 
> In Libraries/ Components/ ToastAndroid/ NativeToastAndroid. js (
> https://github.com/facebook/react-native/pull/24888#discussion_r285207144
> ) :
> 
> > + message: string, + duration: number, + gravity: number, + ) => void; +
> +showWithGravityAndOffset: ( + message: string, + duration: number, +
> gravity: number, + xOffset: number, + yOffset: number, + ) => void; +} +
> +export default (Platform.OS === 'android' + ?
> TurboModuleRegistry.getEnforcing<Spec>('ToastAndroid') + : null);
> 
> Is it odd to export default null ? I wonder if we shoudn't export at all if
> iOS
> 
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub (
> https://github.com/facebook/react-native/pull/24888?email_source=notifications&email_token=AAFEVR4ELLVKD5VF52AUKOTPV3PQTA5CNFSM4HNKWMQ2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBY7VHRQ#pullrequestreview-239031238
> ) , or mute the thread (
> https://github.com/notifications/unsubscribe-auth/AAFEVRZ67WZG6ISPQHEVEFDPV3PQTANCNFSM4HNKWMQQ
> ).
> 
> 
>

ericlewis commented:
its not possible to export *nothing* i.e., make the export itself conditional

Sent via Superhuman ( https://sprh.mn/?vip=ericlewis777@gmail.com )

On Fri, May 17, 2019 at 4:56 PM, Eric Lewis < ericlewis777@gmail.com > wrote:

> 
> you have to export something, undefined or null. i dont think there is a
> specific on witch. 
> 
> 
> Sent via Superhuman ( https://sprh.mn/?vip=ericlewis777@gmail.com )
> 
> 
> 
> On Fri, May 17, 2019 at 12:55 PM, Luna < notifications@ github. com (
> notifications@github.com ) > wrote:
> 
>> 
>> 
>> *@lunaleaps* commented on this pull request.
>> 
>> 
>> 
>> In Libraries/ Components/ ToastAndroid/ NativeToastAndroid. js (
>> https://github.com/facebook/react-native/pull/24888#discussion_r285207144
>> ) :
>> 
>> > + message: string, + duration: number, + gravity: number, + ) => void; +
>> +showWithGravityAndOffset: ( + message: string, + duration: number, +
>> gravity: number, + xOffset: number, + yOffset: number, + ) => void; +} +
>> +export default (Platform.OS === 'android' + ?
>> TurboModuleRegistry.getEnforcing<Spec>('ToastAndroid') + : null);
>> 
>> Is it odd to export default null ? I wonder if we shoudn't export at all if
>> iOS
>> 
>> 
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub (
>> https://github.com/facebook/react-native/pull/24888?email_source=notifications&email_token=AAFEVR4ELLVKD5VF52AUKOTPV3PQTA5CNFSM4HNKWMQ2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBY7VHRQ#pullrequestreview-239031238
>> ) , or mute the thread (
>> https://github.com/notifications/unsubscribe-auth/AAFEVRZ67WZG6ISPQHEVEFDPV3PQTANCNFSM4HNKWMQQ
>> ).
>> 
>> 
>> 
> 
>

wojteg1337 commented:
@ericlewis, I removed the conditional export of turbo module since it's never used in `ios.js` files. In iOS files, I added stubs to return correct types.

wojteg1337 commented:
also I'm not sure what causes `test_android` to fail, could you help me with it?

uqmessias commented on Libraries/Components/ToastAndroid/NativeToastAndroid.js:
```diff
@@ -0,0 +1,44 @@
+  |};
+
+  +show: (message: string, duration: number) => void;
+  +showWithGravity: (
```
Since this is android only, shouldn't it be optional?

uqmessias commented on Libraries/Components/ToastAndroid/NativeToastAndroid.js:
```diff
@@ -0,0 +1,44 @@
+    duration: number,
+    gravity: number,
+  ) => void;
+  +showWithGravityAndOffset: (
```
Ditto

lunaleaps commented:
@wojteg1337 do you have the verbose output of the failing test?

wojteg1337 commented:
@lunaleaps when I add a verbose flag to the command from test `node cli.js bundle --max-workers 2 --platform android --dev true --entry-file ReactAndroid/src/androidTest/js/TestBundle.js --bundle-output ReactAndroid/src/androidTest/assets/AndroidTestBundle.js --verbose` the output seems to be the same as the one from CI
